### PR TITLE
book: added generating CRD documentation

### PR DESCRIPTION
**Pull request by droot**
Renamed 'Using Annotations' section to 'Generating CRD'
Cleaned up the previous docs for annotations
Added docs for 'adding printer columns' feature

### CODE REVIEW

File: docs/book/beyond_basics/generating_crd.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	// +kubebuilder:validation:Enum=1,2,3
+	Rank    int    `json:"rank"`
+}
+
+```
+
+## Additional printer columns
+
+Starting with Kubernetes 1.11, kubectl uses [server-side
+printing](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns).
```
Comment (mengqiy) [inline docs/book/beyond_basics/generating_crd.md]: It says `server-side printing` but points to `CRD additional printer columns feature`  document.
I'd suggest removing the link in this sentence and moving the link to the later sentence:
```
You can [customize these columns using a CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns).
```

File: docs/book/beyond_basics/generating_crd.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	Spec   CronTabSpec   `json:"spec,omitempty"`
+	Status CronTabStatus `json:"status,omitempty"`
+}
+
+```
+
+
+## Subresource
+Custom resources supports `/status` and `/scale` subresources. You can learn
```
Comment (mengqiy) [inline docs/book/beyond_basics/generating_crd.md]: We may want to say
```
Custom resources **only** supports `/status` and `/scale` subresources by 1.13 release.
```

File: docs/book/beyond_basics/generating_crd.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+To enable `/status` subresource, annotate the kind with `// +kubebuilder:subresource:status` format.
+
+### 2. Scale
+To enable `/scale` subresource, annotate the kind with `// +kubebuilder:subresource:scale:specpath=<jsonpath>,statuspath=<jsonpath>,selectorpath=<jsonpath>` format.
+
+Scale subresource annotation contains three fields: `specpath`, `statuspath` and `selectorpath`.
+
+- `specpath` refers to `specReplicasPath` attribute of Scale object, and value `jsonpath` defines the JSONPath inside of a custom resource that corresponds to `Scale.Spec.Replicas`. This is a required field.
+- `statuspath` refers to `statusReplicasPath` attribute of Scale object. and the `jsonpath` value of it defines the JSONPath inside of a custom resource that corresponds to `Scale.Status.Replicas`. This is a required field.
+- `selectorpath` refers to `labelSelectorPath` attribute of Scale object, and the value `jsonpath` defines the JSONPath inside of a custom resource that corresponds to Scale.Status.Selector. This is an optional field.
```
Comment (mengqiy) [inline docs/book/beyond_basics/generating_crd.md]: Quote the path
```
`Scale.Status.Selector`
```

File: docs/book/beyond_basics/generating_crd.md (hunk 1 above)
Comment (droot) [reply docs/book/beyond_basics/generating_crd.md]: Done.

File: docs/book/beyond_basics/generating_crd.md (hunk 2 above)
Comment (droot) [reply docs/book/beyond_basics/generating_crd.md]: Done.

File: docs/book/beyond_basics/generating_crd.md (hunk 3 above)
Comment (droot) [reply docs/book/beyond_basics/generating_crd.md]: Done.

**Comment by droot** [comment]
Thanks for the review @mengqiy . Addressed your comments. PTAL.

**Pull request merged**